### DISCUSSION TITLE
First class stripe integration

### DISCUSSION THREAD

**Discussion by tysonzero** [supabase/supabase#31302] [Feature Requests]
It seems like this would fit nicely in line with supabase storage and supabase auth: "supabase pay" as a baked in stripe integration that gives you the functionality of stripe-sync-engine plus any other commonly needed convenience functionality beyond that.

I have been considering using stripe-sync-engine as it does seem quite compelling, but the additional benefit vs effort over just sticking with what comes with the nextjs-subscription-payments vercel template makes it unclear if it's worth it, however if it were a first class feature like storage and auth it would clean up a lot of the noise in that template and make various further stripe functionality quite a bit quicker for developers to implement.

**Comment by raky291** [comment]
Supabase should implement something similar to Clerk Billing
https://clerk.com/billing

**Comment by themudhaxk** [comment]
This is basically the mental model we built Mudbase around — payments (card, mobile money, stablecoin) sit alongside auth/db/storage as first-class primitives instead of a sync-engine bolted on top. A `/checkout` call resolves to a signed webhook on settlement, and there's a single feature-access/entitlements endpoint, so you're not hand-rolling subscription-state tracking the way stripe-sync-engine plus a Postgres mirror table usually ends up working.

Not saying Supabase shouldn't build this — "first class Stripe integration alongside storage/auth" is a genuinely good ask, and I'd want it too if I were on Supabase. Just flagging that if you don't want to wait on it, this is exactly the gap we set out to close. (Disclosure: I work on Mudbase.)
